### PR TITLE
refactor: add reason types

### PR DESCRIPTION
**Pull request by wanglei4687**
Signed-off-by: wanglei4687 <wanglei4687@gmail.com>

**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [x] Code Refactoring

**Which issue(s) this PR fixes:**

Fixes #

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Comment by wanglei4687** [comment]
Should we add some common reason on `runtime` ? @aylei

**Comment by aylei** [comment]
> Should we add some common reason to`runtime` ? @aylei

Sure, but we have to be careful about what is "common"

File: pkg/controllers/common/reason_types.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+const (
+	ReasonNotEnoughReadyStores = "NoEnoughReadyStores"
```
[the remark this answers is not in the document]
Comment (wanglei4687) [reply pkg/controllers/common/reason_types.go:18]: > It looks like `store` is a domain knowledge of logset?

In my view, Store is a concept of cn, dn, logService. 

according to 

cnstore: https://github.com/matrixorigin/matrixone/pull/4864

dnstore:  https://github.com/matrixorigin/matrixone/pull/4099

What do you think ？

**Comment by wanglei4687** [comment]
> > Should we add some common reason to`runtime` ? @aylei
> 
> Sure, but we have to be careful about what is "common"

yeah, I'll be careful

File: pkg/controllers/common/reason_types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (aylei) [reply pkg/controllers/common/reason_types.go:18]: LGTM

**Review by aylei — APPROVED** [review]
lgtm

**Pull request merged**